lib/next_rails/gem_info.rb

```diff
@@ -77,14 +77,14 @@ def from_rails?
     end
 
     def state(rails_version)
-      if compatible_with_rails?(rails_version: rails_version)
-        :compatible
+      return :compatible if compatible_with_rails?(rails_version: rails_version)
+
+      if latest_compatible_version.nil?
+        :incompatible
       elsif latest_compatible_version.version == "NOT FOUND"
         :no_new_version
-      elsif latest_compatible_version
-        :found_compatible
       else
-        :incompatible
+        :found_compatible
       end
     end
```

Comment:
From a visual perspective may be, can we make it like this?

```
      if compatible_with_rails?(rails_version: rails_version)
        :compatible
      elsif latest_compatible_version && latest_compatible_version.version == "NOT FOUND"
        :no_new_version
      elsif latest_compatible_version
        :found_compatible
      else
        :incompatible
      end
    end
```

what do you think @sean-dickinson ?

Reply:
Ah yeah that makes more sense than what I did 🙃 . Good call.